cranelift/codegen/src/isa/unwind/winx64.rs

```diff
@@ -219,30 +224,49 @@ impl UnwindInfo {
     ) -> CodegenResult<Self> {
         use crate::isa::unwind::input::UnwindCode as InputUnwindCode;
 
+        let word_size: u32 = unwind.word_size.into();
         let mut unwind_codes = Vec::new();
         for c in unwind.prologue_unwind_codes.iter() {
             match c {
-                InputUnwindCode::SaveRegister { offset, reg } => {
-                    unwind_codes.push(UnwindCode::PushRegister {
-                        offset: ensure_unwind_offset(*offset)?,
-                        reg: MR::map(*reg),
-                    });
-                }
-                InputUnwindCode::StackAlloc { offset, size } => {
-                    unwind_codes.push(UnwindCode::StackAlloc {
-                        offset: ensure_unwind_offset(*offset)?,
-                        size: *size,
-                    });
-                }
-                InputUnwindCode::SaveXmmRegister {
+                InputUnwindCode::SaveRegister {
                     offset,
                     reg,
                     stack_offset,
                 } => {
-                    unwind_codes.push(UnwindCode::SaveXmm {
+                    let reg = MR::map(*reg);
+                    let offset = ensure_unwind_offset(*offset)?;
+                    match reg {
+                        MappedRegister::Int(reg) => {
+                            let push_reg_sequence = if let Some(UnwindCode::StackAlloc {
+                                offset: alloc_offset,
+                                size,
+                            }) = unwind_codes.last()
+                            {
+                                *size == word_size && offset == *alloc_offset && *stack_offset == 0
+                            } else {
+                                false
+                            };
+                            if !push_reg_sequence {
+                                return Err(CodegenError::Unsupported(
+                                    "Unsupported UnwindCode::PushRegister sequence".into(),
+                                ));
+                            }
+                            *unwind_codes.last_mut().unwrap() =
```

Comment:
This doesn't seem right to me on first glance as there can be multiple save register operations (i.e. the last wouldn't always be a stack allocation) and it shouldn't be overwriting the stack allocation unwind code, right?

Reply:
looks like UWOP_PUSH_NONVOL does sp -= 8, so we need to remove StackAlloc (or we can just use `UWOP_SAVE_NONVOL` here ?)

Reply:
`StackAlloc` should describe the adjustment of the stack pointer (e.g. `sub rsp, 0x20`) and should always be present for any function that makes a stack allocation.

Reply:
Resolving this as it now makes sense to me why a `StackAlloc` operation would be in the list preceding a `SaveRegister` operation.